quora-service/src/main/java/com/upgrad/quora/service/business/QuestionBusinessService.java

```diff
@@ -28,4 +35,47 @@ public QuestionEntity createQuestion(QuestionEntity questionEntity, final String
         questionEntity.setUser(userAuthEntity.getUser());
         return questionDao.createQuestion(questionEntity);
     }
+
+    public List<QuestionEntity> getAllQuestionsByUser(final String uuid, final String authorizationToken) throws AuthorizationFailedException, UserNotFoundException{
+
+        UserAuthEntity userAuthEntity = questionDao.getUserAuthToken(authorizationToken);
+        if (userAuthEntity == null) {
+            throw new AuthorizationFailedException("ATHR-001", "'User has not signed in");
+        }
+        if (userAuthEntity.getLogoutAt() != null) {
+            throw new AuthorizationFailedException("ATHR-002", "User is signed out.Sign in first to post a question");
+        }
+        if(userAuthEntity.getUser().getRole().equals("nonadmin") || !userAuthEntity.getUser().equals(questionDao.getQuestionByUuid(uuid).getUser())){
+            throw new AuthorizationFailedException("ATHR-003", "Only the question owner or admin can delete the question");
+
+        }
+        if (userDao.getUserByUuid(uuid)==null) {
+            throw new UserNotFoundException("USR-001", "User with entered uuid to be deleted does not exist");
+        }
+
+        return questionDao.getAllQuestionsByUser(uuid);
+    }
+
```

Comment:
Add this line above `userQuestionDelete` method:
```suggestion
@Transactional(propagation = Propagation.REQUIRED)
```